mobile_verifier/src/reward_shares.rs

```diff
@@ -233,13 +255,14 @@ impl PocShares {
                 .get_average(&heartbeat_reward.hotspot_key)
                 .as_ref()
                 .map_or(Decimal::ZERO, SpeedtestAverage::reward_multiplier);
-            *poc_shares
+            poc_shares
                 .hotspot_shares
                 .entry(heartbeat_reward.hotspot_key.clone())
                 .or_default()
                 .radio_shares
                 .entry(heartbeat_reward.cbsd_id.clone())
-                .or_default() += heartbeat_reward.reward_weight() * speedmultiplier;
+                .or_default()
+                .update(heartbeat_reward, speedmultiplier);
```

Comment:
so are we ever going to update the value of the given key and if so are we just overwriting the speedtest multiplier for that share? does the answer to that change drastically based on if we're talking about wifi rewards and cbrs rewards where we have one collection of rewards per device for wifi radios but a true aggregate of all radio rewards attached to a given gateway for cbrs radios?

Reply:
the update is called once per heartbeat reward.  This will be a single time for CBRS as there should only be a single heartbeat per hotspot/radio combo.  For wifi hotspots there will be N heartbeat rewards here and thus update will be called N times.